src/ImageManipulation.php

```diff
@@ -303,7 +303,7 @@ public function updateURL(&$url)
 
         // Only update if resampled file is a smaller file size
         if ($resampled->getAbsoluteSize() < $this->getAbsoluteSize()) {
-            $url = $resampled->getURL();
+            $url = $resampled->getURL(false);
```

Comment:
This will probably cover a lot of regular images on pages won't it?  Won't this break the existing functionality of automatically granting view permissions on draft images on draft pages that a content editor has access to?

Reply:
We're not meant to be automatically granting access to draft images on draft page ... we had a security issue: https://github.com/silverstripe/silverstripe-assets/commit/fddf889917c4e58d32a3e6f476bddaf3fa595e41

If this line creates this ability, than it's bug. In any case, this method as zero context on who is using the file, so it's the wrong place to make the call about whether we want to grant access to the file.

Reply:
We've got it as a configurable opt-in to grant on FileShortCodeProvider, which is inherited by ImageShortCodeProvider - https://github.com/silverstripe/silverstripe-assets/blob/1/src/Shortcodes/FileShortcodeProvider.php#L52

Wondering if there's any value in doing something similar here

Reply:
My instinct would be no ... the methods here are generic. They are unaware of the wider context they are being called in. 

Also, when we implemented the "configurable session grant" option, session grant was the only way of accessing protected files. Since then, we updated the AssetStore to dynamically check if the current user has access to the file on the request. So the need to make this configurable has been reduce.